modin/logging/logger_function.py

```diff
@@ -17,35 +17,92 @@
 ``logger_decorator`` is used for decorating individual Modin functions.
 """
 
+from typing import Optional, Callable, Union, Type
+from types import FunctionType, MethodType
 from functools import wraps
+from logging import Logger
 
 from modin.config import LogMode
 from .config import get_logger
 
+_MODIN_LOGGER_NOWRAP = "__modin_logging_nowrap__"
 
-def logger_decorator(modin_layer: str, function_name: str, log_level: str):
+
+def disable_logging(func):
     """
-    Log Decorator used on specific internal Modin functions.
+    Disable logging of one particular function. Useful for decorated classes.
 
     Parameters
     ----------
-    modin_layer : str
-        Specified by the logger (PANDAS-API).
-    function_name : str
-        The name of the function the decorator is being applied to.
-    log_level : str
-        The log level (logging.INFO, logging.DEBUG, logging.WARNING, etc.).
+    func : callable
+        A method in a logger-decorated class for which logging should be disabled.
 
     Returns
     -------
     func
-        A decorator function.
+        A function with logging disabled.
     """
+    setattr(func, _MODIN_LOGGER_NOWRAP, True)
+    return func
 
-    def decorator(f):
-        """Decorate function to add logs to Modin API function."""
 
-        @wraps(f)
+def logger_decorator(
+    modin_layer: Union[str, Callable, Type] = "PANDAS-API",
+    name: Optional[str] = None,
+    log_level: Optional[str] = "info",
+):
+    """
+    Log Decorator used on specific internal Modin functions or classes.
+
+    Parameters
+    ----------
+    modin_layer : str or object to decorate, default: "PANDAS-API"
+        Specified by the logger (e.g. PANDAS-API).
+        If it's an object to decorate, call logger_decorator() on it with default arguments.
+    name : str, optional
+        The name of the object the decorator is being applied to.
+        Composed from the decorated object name if not specified.
+    log_level : str, default: "info"
+        The log level (INFO, DEBUG, WARNING, etc.).
+
+    Returns
+    -------
+    func
+        A decorator function.
+    """
+    if not isinstance(modin_layer, str):
+        # assume the decorator is used in a form without parenthesis like:
+        # @logger_decorator
+        # def func()
+        return logger_decorator()(modin_layer)
+
+    log_level = log_level.lower()
+    assert hasattr(Logger, log_level.lower()), f"Invalid log level: {log_level}"
+
+    def decorator(obj):
+        """Decorate function or class to add logs to Modin API function(s)."""
+        if isinstance(obj, type):
+            seen = {}
+            for attr_name, attr_value in vars(obj).items():
+                if isinstance(attr_value, (FunctionType, MethodType)) and not hasattr(
+                    attr_value, _MODIN_LOGGER_NOWRAP
```

Comment:
In the logger_metaclass original code, we exclude `__getattribute__` since that can lead to exceptions being raised. I have attached two log files in the open-source Slack for reference.
```
exclude_attributes = {"__getattribute__"}
        for attribute_name, attribute in class_dict.items():
            if (
                isinstance(attribute, (FunctionType, MethodType))
                and attribute_name not in exclude_attributes
            ):
```

Reply:
Yeah, I've decorated `__getattribute__` on `BasePandasDataset` with `@disable_logging`.

Reply:
I could, but it feels like a hack. I had tried running your example, and I don't see any issues with that script of yours.
Could you please post the output you're seeing?